homeassistant/components/daikin/sensor.py

```diff
@@ -74,3 +99,45 @@ async def async_update(self):
     def device_info(self):
         """Return a device description for device registry."""
         return self._api.device_info
+
+
+class DaikinPowerSensor(DaikinSensor):
+    """Representation of a power/energy consumption sensor."""
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement."""
+        if self._device_attribute == ATTR_TOTAL_POWER:
+            return POWER_KILO_WATT
+        if self._device_attribute in (ATTR_COOL_ENERGY, ATTR_HEAT_ENERGY):
+            return ENERGY_KILO_WATT_HOUR
+        return None
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        if self._device_attribute == ATTR_TOTAL_POWER:
+            return self._api.device.current_total_power_consumption
+        if self._device_attribute == ATTR_COOL_ENERGY:
+            return self._api.device.last_hour_cool_power_consumption
+        if self._device_attribute == ATTR_HEAT_ENERGY:
+            return self._api.device.last_hour_heat_power_consumption
+        return None
+
+
+class DaikinClimateSensor(DaikinSensor):
```

Comment:
I would prefer if the ClimateSensor was added before the PowerSensor (it might help git to reuse more of the code in the diff).

Reply:
I've switched them